Userland/Libraries/LibELF/Validation.cpp

```diff
@@ -192,7 +192,7 @@ bool validate_elf_header(const ElfW(Ehdr) & elf_header, size_t file_size, bool v
     return true;
 }
 
-bool validate_program_headers(const ElfW(Ehdr) & elf_header, size_t file_size, const u8* buffer, size_t buffer_size, String* interpreter_path, bool verbose)
+bool validate_program_headers(const ElfW(Ehdr) & elf_header, size_t file_size, const u8* buffer, size_t buffer_size, char (*interpreter_path_buffer)[PATH_MAX], bool verbose)
```

Comment:
I don't love this, but I can't think of a better way.

Reply:
We could use a StringBuilder preallocated to PATH_MAX by the caller? That might be a tiny improvement?

Reply:
Actually yeah, I hadn't considered that, we can just use a normal StringBuilder, and make it return ErrorOr<bool>, which will allow us to also handle this in Userland in the future.

Reply:
Updated to use ErrorOr<bool> + StringBuilder*